docs/tutorials/07_PDATutorial.py

```diff
@@ -0,0 +1,385 @@
+
+# %%
+# Running the PDA Filter
+# ----------------------
+# With these components, we can run the simulated data and clutter through the Kalman filter.
+
+# Create prior
+from stonesoup.types.state import GaussianState
+prior = GaussianState([[0], [1], [0], [1]], np.diag([1.5, 0.5, 1.5, 0.5]), timestamp=start_time)
+
+# Loop through the predict, hypothesise, associate and update steps.
+from stonesoup.types.track import Track
+from stonesoup.types.array import StateVectors  # For storing state vectors during association
+from stonesoup.functions import gm_reduce_single  # For merging states to get posterior estimate
+from stonesoup.types.update import GaussianStateUpdate  # To store posterior estimate
+
+track = Track([prior])
+for n, measurements in enumerate(all_measurements):
+    hypotheses = data_associator.associate([track],
+                                           measurements,
+                                           start_time + timedelta(seconds=n))
+
+    hypotheses = hypotheses[track]
+
+    # Loop through each hypothesis, creating posterior states for each, and merge to calculate
+    # approximation to actual posterior state mean and covariance.
+    posterior_states = []
+    posterior_state_weights = []
+    for hypothesis in hypotheses:
+        if not hypothesis:
+            posterior_states.append(hypothesis.prediction)
+        else:
+            posterior_state = updater.update(hypothesis)
+            posterior_states.append(posterior_state)
+        posterior_state_weights.append(
+            hypothesis.probability)
+
+    means = StateVectors([state.state_vector for state in posterior_states])
+    covars = np.stack([state.covar for state in posterior_states], axis=2)
+    weights = np.asarray(posterior_state_weights)
+
+    # Reduce mixture of states to one posterior estimate Gaussian.
+    post_mean, post_covar = gm_reduce_single(means, covars, weights)
+
+    # Add a Gaussian state approximation to the track.
+    track.append(GaussianStateUpdate(
+        post_mean, post_covar,
+        hypotheses,
+        hypotheses[0].measurement.timestamp))
+
+
+# %%
+# Plot the resulting track
+ax.plot([state.state_vector[0, 0] for state in track[1:]],  # Skip plotting the prior
+        [state.state_vector[2, 0] for state in track[1:]],
+        marker=".")
+fig
+
+# %%
+# Plotting ellipses representing the gaussian estimate state at each update.
+from matplotlib.patches import Ellipse
+for state in track[1:]:  # Skip the prior
+    w, v = np.linalg.eig(measurement_model.matrix()@state.covar@measurement_model.matrix().T)
+    max_ind = np.argmax(v[0, :])
+    orient = np.arctan2(v[max_ind, 1], v[max_ind, 0])
+    ellipse = Ellipse(xy=state.state_vector[(0, 2), 0],
+                      width=np.sqrt(w[0])*2, height=np.sqrt(w[1])*2,
+                      angle=np.rad2deg(orient),
+                      alpha=0.2)
+    ax.add_artist(ellipse)
+fig
+
+# %%
+# Joint Probabilistic Data Association
```

Comment:
@oharrald-Dstl - I know we discussed have this as one tutorial, but I'm thinking now it's worth splitting into two, as it's a little long like this.